pkg/network/filter/packet_source_linux.go

```diff
@@ -129,6 +130,21 @@ func (p *AFPacketSource) SetBPF(filter []bpf.RawInstruction) error {
 	return p.TPacket.SetBPF(filter)
 }
 
+// getPktType gets whether this packet capture was incoming or outgoing.
+// incoming: unix.PACKET_HOST
+// outgoing: unix.PACKET_OUTGOING
+func getPktType(stats gopacket.CaptureInfo) (uint8, error) {
+	// since we set addPktType = true, AncillaryData will contain an AncillaryPktType element;
+	// however, it might not be the first element, so scan through.
+	for _, data := range stats.AncillaryData {
+		pktType, ok := data.(afpacket.AncillaryPktType)
+		if ok {
+			return pktType.Type, nil
+		}
+	}
+	return 0, errors.New("ancillary data didn't contain AncillaryPktType (make sure addPktType = true)")
```

Comment:
This won't always be there; it depends on an option being set on the packet parser, so we don't have to error out here.

Reply:
I think you can move the loop into `VisitPackets` and just set `pktInfo.PktType` if `AncillaryPktType` is present.